cmd/thermal-recorder/main.go

```diff
@@ -132,25 +135,78 @@ func runMain() error {
 	}
 }
 
+type HeaderInfo struct {
+	resX      int
+	resY      int
+	fps       int
+	framesize int
+	model     string
+	brand     string
+}
+
+func (h *HeaderInfo) ResX() int {
+	return h.resX
+}
+func (h *HeaderInfo) ResY() int {
+	return h.resY
+}
+func (h *HeaderInfo) FPS() int {
+	return h.fps
+}
+
+func NewHeader(headerInfo map[string]interface{}) *HeaderInfo {
+	return &HeaderInfo{
+		resX:      headerInfo[headers.XResolution].(int),
+		resY:      headerInfo[headers.YResolution].(int),
+		fps:       headerInfo[headers.FPS].(int),
+		framesize: headerInfo[headers.FrameSize].(int),
+		model:     headerInfo[headers.Model].(string),
+		brand:     headerInfo[headers.Model].(string),
+	}
+}
+
+func readHeader(conn net.Conn) (*HeaderInfo, error) {
+	var buf bytes.Buffer
+	scanner := bufio.NewScanner(conn)
+	for {
+		if ok := scanner.Scan(); !ok {
+			break
+		}
+		line := scanner.Text()
+		if strings.Trim(line, " ") == "" {
+			break
+		}
+		buf.WriteString(line + "\n")
+	}
```

Comment:
I think I've led you astray here. It looks like Scanner will read chunks of the input, beyond what is required for each match, into an internal buffer. The assumption is you'll keep using the Scanner to read until EOF. This means that it could end up reading in and caching frame data while reading the headers.

A safer approach would be to wrap the `net.Conn` with a `bufio.Reader` and then use that from then on to read from. It has a `ReadString` method which can be used to extract the header lines.

A safer

Reply:
Good to know.